filebeat/module/traefik/access/test/test.log-expected.json

```diff
@@ -2,204 +2,218 @@
     {
         "@timestamp": "2017-10-02T20:22:07.000Z", 
         "event.dataset": "access", 
+        "event.duration": 2000000, 
         "event.module": "traefik", 
+        "http.request.method": "GET", 
+        "http.request.referrer": "http://example.com/login", 
+        "http.response.status_code": 304, 
+        "http.version": "1.1", 
         "input.type": "log", 
         "log.offset": 0, 
+        "source.address": "192.168.33.1", 
+        "source.ip": "192.168.33.1", 
         "traefik.access.backend_url": "http://172.19.0.3:5601", 
         "traefik.access.body_sent.bytes": 0, 
         "traefik.access.duration": 2, 
         "traefik.access.frontend_name": "Host-host-1", 
-        "traefik.access.http_version": "1.1", 
-        "traefik.access.method": "GET", 
-        "traefik.access.referrer": "http://example.com/login", 
-        "traefik.access.remote_ip": "192.168.33.1", 
         "traefik.access.request_count": 262, 
-        "traefik.access.response_code": "304", 
-        "traefik.access.url": "/ui/favicons/favicon-16x16.png", 
-        "traefik.access.user_agent.device": "Other", 
-        "traefik.access.user_agent.major": "61", 
-        "traefik.access.user_agent.minor": "0", 
-        "traefik.access.user_agent.name": "Chrome", 
-        "traefik.access.user_agent.original": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36", 
-        "traefik.access.user_agent.os": "Linux", 
-        "traefik.access.user_agent.os_name": "Linux", 
-        "traefik.access.user_agent.patch": "3163", 
         "traefik.access.user_identifier": "-", 
-        "traefik.access.user_name": "-"
+        "url.original": "/ui/favicons/favicon-16x16.png", 
+        "user.name": "-", 
+        "user_agent.device": "Other", 
+        "user_agent.name": "Chrome", 
+        "user_agent.original": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/61.0.3163.100 Safari/537.36", 
+        "user_agent.os.full_name": "Linux", 
+        "user_agent.os.name": "Linux", 
+        "user_agent.os.version": "..", 
```

Comment:
?

Reply:
Yes, this is what the second caveat is about. Right now the UA parser gives us major/minor/patch most of the time, and doesn't give us a full version string.

So I'm setting `user_agent.version` and `user_agent.os.version` ([here](https://github.com/elastic/beats/pull/9005/files#diff-5f2f0d705013b636262f7085a1bd369cR112) and [here](https://github.com/elastic/beats/pull/9005/files#diff-5f2f0d705013b636262f7085a1bd369cR101) respectively) with a trivial interpolation. To actually do this right, ideally IN would give this back to us as a full version string (may even be better wrt test versions like betas), second best approach would be to put together a small reuseable pipeline that rebuilds the version string based on these 3-4 fields that may or may not be there, and leverage that everywhere at once.

And the worst but simplest approach was simply to interpolate directly there. Even if I guard it with an if on the presence of `.major`, we'll have broken-looking version numbers whenever another one of the numbers is missing ([like here](https://github.com/elastic/beats/pull/9005/files#diff-932723c2076c2bd3dd6d24df4628a161R99)). So I didn't want to waste a bunch of time on this.

Another alternative for this PR is simply to address this later, and leave major/minor/patch in place for now, like all of the other access log modules.

Reply:
SGTM to delay this.